sdk/go/common/util/rpcutil/health.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2016-2022, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rpcutil
+
+import (
+	"context"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	healthpb "google.golang.org/grpc/health/grpc_health_v1"
+	"google.golang.org/grpc/status"
+)
+
+// Healthcheck will poll the address at duration intervals and then call cancel once it reports unhealthy
+func Healthcheck(context context.Context, addr string, duration time.Duration, cancel context.CancelFunc) error {
+	conn, err := grpc.Dial(
+		addr,
+		grpc.WithInsecure(),
+		grpc.WithUnaryInterceptor(OpenTracingClientInterceptor()),
```

Comment:
Interesting. For most analysis I'd want to do on the trace, having these calls in the trace wouldn't be that helpful. But it seems reasonable to include it for completeness. On the fence about this.

Reply:
Better to trace than not to, IMO. Might help solve why a provider is failing if the trace shows something strange in healthcheck. Unlikely, but possible?